src/briefcase/platforms/android/gradle.py

```diff
@@ -229,22 +230,51 @@ def run_app(self, app: BaseConfig, device_or_avd=None, **kwargs):
 
         # To start the app, we launch `org.beeware.android.MainActivity`.
         with self.input.wait_bar("Launching app..."):
+            # Any log after this point must be associated with the new instance
+            # Pad by a few seconds because the android emulator's clock and the
+            # local system clock may not be perfectly aligned.
+            start_time = datetime.datetime.now() - datetime.timedelta(seconds=10)
             adb.start_app(package, "org.beeware.android.MainActivity")
             pid = None
-            while not pid:
-                pid = adb.pidof(package)
+            attempts = 0
+            delay = 0.01
+            while not pid and attempts < 500:
```

Comment:
This should check the elapsed time rather than count the attempts, because running adb might take significantly longer than 0.01 seconds.